lib/OAuth/Cmdline/MicrosoftOnline.pm

```diff
@@ -22,46 +52,63 @@ OAuth::Cmdline::MicrosoftOnline - Microsoft Online-specific settings for OAuth::
 
 =head1 SYNOPSIS
 
-    my $oauth = OAuth::Cmdline::MicrosoftOnline->new( );
+    my $oauth = OAuth::Cmdline::MicrosoftOnline->new( resource => "https://graph.microsoft.com", ... );
     $oauth->access_token();
 
 =head1 DESCRIPTION
 
 This class overrides methods of C<OAuth::Cmdline> if Microsoft Online's Web API 
 requires it.
 
+The parameter 'resource' is mandatory, and is poorly described at L<https://msdn.microsoft.com/en-us/library/azure/dn645542.aspx>. It tells the OAuth API what protected resource you are trying to access. For example, to access Azure Graph (to manage user accounts in Azure AD etc.), the correct resource URI is C<https://graph.microsoft.com>. A URI does not have to be a URL, but Microsoft choose to use URLs for their URIs, so if you are trying to access a different endpoint protected by the Microsoft Online OAuth system, then it will probably look like a URL.
```

Comment:
consider splitting it on multiple lines, easier for maintenance and should not impact the pod itself